src/index.js

```diff
@@ -79,9 +83,42 @@ export default class MarkersPlugin extends UICorePlugin {
       internalMarker.emitter.off("tooltipChanged", internalMarker.tooltipChangedHandler)
     }
     internalMarker.onDestroy()
+    this._markers.splice(index, 1)
     return true
   }
 
+  /*
+   * Clear all existing markers
+   */
+  clearMarkers() {
+    let self = this
+    if (!this._markers) {
+      return
+    } 
+    // Create a copy of the array since you cannot iterate and remove all its members correctly at the same time
+    let markersCopy = this._markers.slice(0)
+    markersCopy.forEach((marker, index) => {
+      self._removeInternalMarker(marker, index)
+    })
+  }
+
+  /*
+   * Get all markers
+   */
+  getAll() {
```

Comment:
This is a good method but the contents of `this._markers` is meant to remain internal. It would be better to return the `source` property values to the user gets back what they provided (and would use to remove).

I think `map` is pretty well supported now but it might be worth checking.

```javascript
return this._markers.map(internalMarker => {
  return internalMarker.source
})